### PR TITLE
feat: validate signup payload

### PR DESCRIPTION
**Pull request by AlexKMarshall**
Validates the signup form with valibot. Email must be an email, password must be more than 8 chars. We don't yet trigger an action signup in the database

## Greptile Summary

This PR introduces comprehensive server-side form validation for the signup functionality using the Valibot validation library. The implementation adds robust input validation to ensure emails are properly formatted and passwords meet a minimum length requirement of 8 characters.

The changes follow React Router's action-based form handling pattern by integrating validation logic directly into the signup action function. When validation fails, the action returns structured error data containing the original payload and flattened validation issues. When validation succeeds, it logs the validated data and returns a success response.

Valibot was chosen as the validation library for its TypeScript-first approach and lightweight nature (~2KB vs Zod's ~12KB). The library integrates seamlessly with the existing React Router setup and provides type-safe validation schemas. The validation schema uses Valibot's object schema with piped validators for email format checking and minimum password length enforcement.

The PR enhances user experience by implementing proper error display with accessibility features including `aria-invalid` and `aria-describedby` attributes. The form includes `noValidate` to disable browser validation in favor of server-side validation. Additionally, comprehensive test coverage includes both unit tests for the action function and Storybook stories for visual testing of success and error states.

A separate Node.js test configuration was added to support server-side unit testing, distinguishing it from existing browser-based Storybook tests. This establishes a solid foundation for authentication security and data integrity validation.

## Important Files Changed

<details>
<summary>Files Modified</summary>

| Filename | Score | Overview |
|----------|-------|----------|
| `package.json` | 5/5 | Added valibot validation library and faker.js for test data generation |
| `vitest.config.ts` | 5/5 | Added Node.js unit test configuration for server-side validation testing |
| `app/auth/signup.tsx` | 4/5 | Implemented comprehensive server-side form validation with error display and accessibility features |
| `app/auth/signup.test.ts` | 4/5 | Added unit tests for signup action validation with comprehensive edge case coverage |
| `app/auth/signup.stories.tsx` | 4/5 | Updated Storybook stories to include validation error states and accessibility testing |

</details>

## Manual Testing Recommendations

- [ ] Test form submission with valid email and password (>8 characters)
- [ ] Test form submission with invalid email formats (missing @, invalid domain)
- [ ] Test form submission with passwords under 8 characters
- [ ] Test form submission with empty fields
- [ ] Verify accessibility features work with screen readers
- [ ] Confirm error messages display correctly and are accessible
- [ ] Test keyboard navigation through form and error states
- [ ] Verify form state persists on validation errors

## Automated Testing

- [x] Unit tests for signup action validation logic
- [x] Storybook visual tests for error states
- [x] Accessibility testing in Storybook
- [ ] Integration tests for complete form submission flow
- [ ] Performance tests for validation logic with large payloads
- [ ] Bundle size impact analysis for valibot addition
- [ ] Tests for concurrent form submissions

## Architecture and Design Review

**Validation Strategy**: Server-side validation with valibot follows security best practices and provides excellent TypeScript integration. The choice of valibot over Zod demonstrates consideration for bundle size optimization.

**Error Handling**: Structured error response format enables granular field-level error display while preserving user input, following modern form UX patterns.

**Accessibility**: Proper ARIA attributes (`aria-invalid`, `aria-describedby`) enhance screen reader compatibility and demonstrate attention to inclusive design principles.

**Testing Architecture**: Excellent separation between Node.js unit tests and browser-based Storybook tests enables comprehensive coverage across different environments.

**Areas for Enhancement**: Replace console.log statements with proper logging infrastructure and consider adding rate limiting for production security.

## Confidence score: 4/5

• This PR is safe to merge with solid validation implementation and comprehensive testing coverage
• The validation logic is well-structured and follows best practices, with proper error handling and accessibility features
• Minor attention needed for `app/auth/signup.tsx` to replace console.log statements, though current implementation is robust

## Sequence Diagram
```mermaid
sequenceDiagram
    participant User
    participant SignupForm as Signup Form
    participant Action as signup action()
    participant Valibot as Valibot Schema
    
    User->>SignupForm: Fills form (email, password)
    User->>SignupForm: Clicks "Sign Up" button
    SignupForm->>Action: POST form data
    Action->>Action: Extract formData to payload
    Action->>Valibot: v.safeParse(SignupSchema, payload)
    
    alt Validation fails
        Valibot-->>Action: Return validation issues
        Action-->>SignupForm: Return {result: "error", payload, issues}
        SignupForm->>SignupForm: useActionData() receives error data
        SignupForm-->>User: Display form with validation errors
    else Validation succeeds
        Valibot-->>Action: Return validated output
        Action-->>SignupForm: Return {result: "success", payload, issues: null}
        SignupForm->>SignupForm: useActionData() receives success data
        SignupForm-->>User: Display success state (no errors)
    end
```

### CODE REVIEW

File: app/auth/signup.test.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    issues: null,
+  })
+})
+
+test('missing email', async () => {
+  const request = makeSignupRequest({ email: '' })
+
+  const result = await callSignupAction(request)
+
+  expect(result).toEqual(expect.objectContaining({ result: 'error' }))
```
[the remark this answers is not in the document]
Comment (AlexKMarshall) [reply app/auth/signup.test.ts:67]: Good suggestion. I'll do this in a follow up PR when I add better error copy, not just the valibot default messages
